components/script/dom/readablestream.rs

```diff
@@ -838,6 +898,74 @@ impl Callback for SourceCancelPromiseRejectionHandler {
     }
 }
 
+#[derive(Clone, JSTraceable, MallocSizeOf)]
+struct ForwardPromiseFulfillmentHandler {
+    #[conditional_malloc_size_of]
+    target: Rc<Promise>,
```

Comment:
Can we use a more specific name than target? There are several layers of indirection here that I'm having trouble holding in my brain.